query/security/ldap/src/main/java/org/codice/ddf/admin/ldap/commons/ServerGuesser.java

```diff
@@ -329,4 +365,25 @@ private OpenDjGuesser(Connection connection) {
       super(connection);
     }
   }
+
+  private static class NameSorter implements Comparator<String> {
```

Comment:
✏️ This Comparator needs javadoc. I had to scroll back up to see how the `preference` field was used in order to realize that for members, rather than groups, your query conditions will allow users through that do not include the `preference` String and the point is to sort those to the end.

Reply:
Good point.  I'll add some javadoc